crates/json-abi/src/item.rs

```diff
@@ -356,7 +358,95 @@ impl AbiItem<'_> {
     }
 }
 
+impl FromStr for Constructor {
+    type Err = ParserError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::parse(s)
+    }
+}
+
+impl Constructor {
+    /// Parses a Solidity event signature string:
+    /// `constructor($($inputs),*) $(anonymous)?`
+    ///
+    /// Note:
+    /// - the name must always be `constructor`.
+    /// - that [`state_mutability`](Self::state_mutability) is not parsed from
+    ///   the input and is always set to [`StateMutability::Payable`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use alloy_json_abi::{Constructor, Param, StateMutability};
+    /// assert_eq!(
+    ///     Constructor::parse("constructor(uint foo, address bar)"),
+    ///     Ok(Constructor {
+    ///         inputs: vec![
+    ///             Param::parse("uint foo").unwrap(),
+    ///             Param::parse("address bar").unwrap()
+    ///         ],
+    ///         state_mutability: StateMutability::Payable,
+    ///     }),
+    /// );
+    /// ```
+    #[inline]
+    pub fn parse(s: &str) -> ParserResult<Self> {
+        parse_sig::<false>(s).map(|(_, inputs, _, _)| Self {
+            inputs,
+            state_mutability: StateMutability::Payable,
+        })
+    }
+
+    /// Computes this constructor's signature: `constructor($($inputs),*)`.
+    ///
+    /// This is the preimage input used to [compute the
+    /// selector](Self::selector).
+    #[inline]
+    pub fn signature(&self) -> String {
+        signature("constructor", &self.inputs, None)
+    }
+
+    /// Computes this constructor's selector: `keccak256(self.signature())[..4]`
+    #[inline]
+    pub fn selector(&self) -> Selector {
+        selector(&self.signature())
+    }
```

Comment:
Idk how useful this is on `Constructor`s

Reply:
am indifferent, cc @prestwich

Reply:
not useful at all. constructors are never called as functions